### PR TITLE
add go mod for pulsar-client-go

### PR DESCRIPTION
**Pull request by wolfstudy**
Signed-off-by: xiaolong.ran <ranxiaolong716@gmail.com>

add `go mod` for project, you can run

```
go mod download
``` 

get the corresponding dependencies.

### CODE REVIEW

**Comment by wolfstudy** [comment]
@merlimat PTAL

**Comment by wolfstudy** [comment]
When we execute `go mod init`, the path used is: `github.com/apache/pulsar-client-go`, so you may need to update your local folder path as follows:

```
mkdir -p ~/github.com/apache/pulsar-client-go
```

File: perf/perf-producer.go [hunk 1]
```diff
@@ -22,11 +22,12 @@ package main
 import (
 	"context"
 	"encoding/json"
+	`github.com/Anankke/bilibili-backup/library/net/http/blademaster/middleware/rate`
```
Comment (merlimat) [inline perf/perf-producer.go]: Is this import intended?

**Comment by merlimat** [comment]
@wolfstudy Can you also add instructions in the top-level README.md on how to compile with dependencies? Would that apply only to people trying to do dev work on the library, right?

**Comment by merlimat** [comment]
Actually, let me merge this, so it will simplify adding the unit tests scripts.

File: perf/perf-producer.go (hunk 1 above)
Comment (wolfstudy) [reply perf/perf-producer.go]: oh no, actually, we shouldn't import this, next pull request, i will remove it

Comment (merlimat) [reply perf/perf-producer.go]: No worries. I already removed it in my code.

**Pull request merged**